client_handlers.go

```diff
@@ -76,14 +88,20 @@ func clientHandshakeHandler(c *Conn) error {
 
 		case *handshakeMessageServerHello:
 			for _, extension := range h.extensions {
-				if e, ok := extension.(*extensionUseSRTP); ok {
+				switch e := extension.(type) {
+				case *extensionUseSRTP:
 					profile, ok := findMatchingSRTPProfile(e.protectionProfiles, c.localSRTPProtectionProfiles)
 					if !ok {
 						return fmt.Errorf("Server responded with SRTP Profile we do not support")
 					}
 					c.state.srtpProtectionProfile = profile
+				case *extensionUseExtendedMasterSecret:
+					c.state.extendedMasterSecret = true
 				}
 			}
+			if c.clientRequireExtendedMasterSecret == false && c.state.extendedMasterSecret == false {
+				return fmt.Errorf("Client required Extended Master Secret extension, but server does not support it")
```

Comment:
Based on the error message here I would've expected a conditional of `c.clientRequireExtendedMasterSecret`, not a comparison to `false`?

Reply:
Whops good catch, thanks! This is now fixed